### ISSUE TITLE
wrong attention masks?

### ISSUE THREAD

**Issue by dhansmair** [lucidrains/flamingo-pytorch#6]
https://github.com/lucidrains/flamingo-pytorch/blob/44920f4191ba3c280ff84c6ebc76025656d1dab5/flamingo_pytorch/flamingo_pytorch.py#L159

In the flamingo paper, the language features in the gated cross-attention layers only attend to the visual features from the immediate preceding image. I believe your attention masks are created in such a way that they attend to the visual features from *all* preceding images. Can you confirm? If so, a fix would be to simply change the '>=' to '=='.

**Comment by lucidrains** [comment]
oh no way! can you screenshot and share the relevant passage? wouldn't it make sense to see all preceding images before answering a question?
